api/tests/integration/domain/usecases/create-and-reconcile-user-to-schooling-registration-by-token_test.js

```diff
@@ -0,0 +1,181 @@
+        const externalUser = {
+          firstName: 'Saml',
+          lastName: 'Jackson',
+        };
+        const token = tokenService.createTokenForStudentReconciliation(externalUser);
+
+        // when
+        const error = await catchErr(createAndReconcileUserToSchoolingRegistrationByToken)({ campaignCode, token, tokenService, campaignRepository });
+
+        // then
+        expect(error).to.be.instanceof(ObjectValidationError);
+      });
+    });
+
+  });
+
+  context('When no schoolingRegistration found', () => {
+
+    let campaignCode;
+    let token;
+
+    beforeEach(async () => {
+      campaignCode = databaseBuilder.factory.buildCampaign().code;
+      await databaseBuilder.commit();
+      const externalUser = {
+        firstName: 'Saml',
+        lastName: 'Jackson',
+        samlId: 'samlId',
+      };
+      token = tokenService.createTokenForStudentReconciliation(externalUser);
+    });
+
+    it('should throw a Not Found error', async () => {
+      // given
+      const birthdate = '2008-01-01';
+
+      // when
+      const error = await catchErr(createAndReconcileUserToSchoolingRegistrationByToken)({
+        campaignCode, token, birthdate,
+        campaignRepository, tokenService, userReconciliationService, schoolingRegistrationRepository
+      });
+
+      // then
+      expect(error).to.be.instanceof(NotFoundError);
+      expect(error.message).to.equal('There are no schooling registrations found');
+    });
+  });
+
+  context('When a schoolingRegistration match the token data and birthdate', () => {
+
+    const firstName = 'Saml';
+    const lastName = 'Jackson';
+    const samlId = 'SamlId';
+
+    let campaignCode;
+    let organizationId;
+    let token;
+
+    beforeEach(async () => {
+      organizationId = databaseBuilder.factory.buildOrganization().id;
+      campaignCode = databaseBuilder.factory.buildCampaign({ organizationId }).code;
+      await databaseBuilder.commit();
+
+      const externalUser = { firstName, lastName, samlId };
+      token = tokenService.createTokenForStudentReconciliation(externalUser);
+    });
+
+    it('should create and reconcile user to schoolingRegistration by token', async () => {
+      // given
+      const schoolingRegistration = databaseBuilder.factory.buildSchoolingRegistration({ firstName, lastName, organizationId });
+      schoolingRegistration.userId = undefined;
+      await databaseBuilder.commit();
+
+      // when
+      const user = await createAndReconcileUserToSchoolingRegistrationByToken({ campaignCode, token,
+        birthdate: schoolingRegistration.birthdate, campaignRepository, tokenService, schoolingRegistrationRepository,
+        userRepository, userReconciliationService, obfuscationService });
+
+      // then
+      expect(user.firstName).to.equal(firstName);
+      expect(user.lastName).to.equal(lastName);
+      expect(user.samlId).to.equal(samlId);
+      expect(user.password).to.be.empty;
+      expect(user.cgu).to.be.false;
```

Comment:
```suggestion
      expect(user.firstName).to.equal(firstName);
      expect(user.lastName).to.equal(lastName);
      expect(user.samlId).to.equal(samlId);
      expect(user.cgu).to.be.false;
```

Reply:
Pourquoi enlever le test sur le `password` vide ? je le trouve assez important car il y a une contrainte NOT NULL en base (c'est la raison pour laquelle on le remplit) et on voit bien ici que l'on ne cherche pas à lui définir un mot de passe.